### ISSUE TITLE
Background in the editor

### ISSUE THREAD

**Issue by maDU59** [ScratchAddons/ScratchAddons#1944]
Hello, I think it can be cool, if you can choose a background for the editor. On scratch, in the editor, the background is white, but de Fanny change it, you can just have a black background with the dark mod... If I want m'y dog, I can't

**Comment by Genius398** [comment]
Yes. The editor looks "LAME" without a background. Maybe a background for the blocks area also?

**Comment by mxmou** [comment]
> Yes. The editor looks "LAME" without a background. Maybe a background for the blocks area also?

Maybe the workspace could be transparent?
![Screenshot](https://user-images.githubusercontent.com/51849865/112122653-cc26ab00-8bc0-11eb-867d-9e70d6ef215a.png)

**Comment by mxmou** [comment]
The only problem with this addon is how the image would be stored. It isn't possible to store large files in `chrome.storage.sync`.

**Comment by maDU59** [comment]
> > Yes. The editor looks "LAME" without a background. Maybe a background for the blocks area also?
> 
> Maybe the workspace could be transparent?

Yes

**Comment by maDU59** [comment]
> The only problem with this addon is how the image would be stored. It isn't possible to store large files in `chrome.storage.sync`.

I don't know. I am not English, I am French, so I just help for the translation on ScratchAddons

**Comment by RedGuy12** [comment]
> The only problem with this addon is how the image would be stored. It isn't possible to store large files in `chrome.storage.sync`.

it could be uploaded via assets.scratch.mit.edu

**Comment by Genius398** [comment]
> > The only problem with this addon is how the image would be stored. It isn't possible to store large files in `chrome.storage.sync`.
> 
> it could be uploaded via assets.scratch.mit.edu

What's that?

**Comment by Genius398** [comment]
The picture should ONLY fit the blocks area (where we drag the blocks).

**Comment by maDU59** [comment]
> The picture should ONLY fit the blocks area (where we drag the blocks).

And where the blocks are

**Comment by Genius398** [comment]
> > The picture should ONLY fit the blocks area (where we drag the blocks).
> 
> And where the blocks are

Yes

**Comment by Genius398** [comment]
The tag "<input type="file" causes the browser to show a text field for the filename and a browse button (if the browser supports file uploads). Two Javascript functions are called. "DoPreview()" is called by the button and it loads the file into an Image object in Javascript and assigns it to the img tag.

**Comment by ErrorGamer2000** [comment]
> The only problem with this addon is how the image would be stored. It isn't possible to store large files in `chrome.storage.sync`.

Maybe add an upload btn then once an image is uploaded do
```js
let src = URL.createObjectUrl(e.target.files[0]);

document.querySelector("#bgimg").src = src
```

Then use [lz-string](https://pieroxy.net/blog/pages/lz-string/index.html) and compress it for storage?

```js
chrome.storage.sync.set("bgimg", LZString.compress(src));
...
document.onload = () => {
    document.querySelector("#bgimg").src = LZString.decompress(chrome.storage.,sync.get(["bgimg"]))
}
```
or something like that . . .
